### PR TITLE
DRA API: check "AdminAccess in use" only once

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This is simpler and an opportunity to explain the concept.

#### Special notes for your reviewer:

Follow-up for https://github.com/kubernetes/kubernetes/pull/127266#discussion_r1821678195

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @thockin

### CODE REVIEW

File: pkg/registry/resource/resourceclaim/strategy.go [hunk 1]
```diff
@@ -182,9 +182,18 @@ func dropDisabledDRAAdminAccessFields(newClaim, oldClaim *resource.ResourceClaim
...
+		// adding a status with AdminAccess=true is allowed. The
+		// scheduler typically doesn't do that (it also checks the
+		// feature gate and refuses to schedule), but the apiserver
+		// would allow it.
+		return
+	}
 
 	for i := range newClaim.Spec.Devices.Requests {
-		if newClaim.Spec.Devices.Requests[i].AdminAccess != nil && !draAdminAccessFeatureInUse(oldClaim) {
+		if newClaim.Spec.Devices.Requests[i].AdminAccess != nil {
```
Comment (thockin) [inline pkg/registry/resource/resourceclaim/strategy.go:196]: why bother checking?

Comment (pohly) [reply pkg/registry/resource/resourceclaim/strategy.go]: For no good reason... [removed](https://github.com/kubernetes/kubernetes/compare/8687cab5aa12baa4775e393a1df9addfb00991fe..d1f0d24ef927b0ff0813d10e5c0716d6caf214df).

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**